Source/WebGPU/WebGPU/ShaderModule.mm

```diff
@@ -239,9 +239,24 @@ auto compilationMessageData(convertMessages(
         m_library.label = label;
 }
 
-id<MTLFunction> ShaderModule::getNamedFunction(const String& name) const
+id<MTLFunction> ShaderModule::getNamedFunction(const String& name, const HashMap<String, float>& keyValueReplacements) const
 {
-    return [m_library newFunctionWithName:name];
+    if (!keyValueReplacements.size())
```

Comment:
What about validation? What happens if there are some pipeline-overridable constants which must be filled in, but they aren't?

Reply:
per the [Metal specification](https://developer.apple.com/documentation/metal/mtllibrary/1515524-newfunctionwithname) we should return an invalid pipeline if we try to use a MTLFunction to create a render pipeline and don't specify the function constants:

```
    id<MTLRenderPipelineState> renderPipelineState = [m_device newRenderPipelineStateWithDescriptor:mtlRenderPipelineDescriptor options:MTLPipelineOptionArgumentInfo reflection:&reflection error:nil];
    if (!renderPipelineState)
        return RenderPipeline::createInvalid(*this);
```

or if we don't specify some of them, then `-[MTLLibrary newFunctionWithName:constantValues: error:]`  would return nil and we and invalid pipeline when `getNamedFunction` returns.

https://developer.apple.com/documentation/metal/mtllibrary/1640020-newfunctionwithname

We can perform our own validation on top of the validation the Metal API does but it is probably not necessary?